frontend/src/metabase/query_builder/components/VisualizationError.jsx

```diff
@@ -22,6 +23,49 @@ const EmailAdmin = () => {
   );
 };
 
+function adjustPositions(error, origSql) {
+  /* Positions in error messages are borked coming in for Postgres errors.
+   * Previously, you would see "blahblahblah bombed out, Position: 119" in a 10-character invalid query.
+   * This is because MB shoves in 'remarks' into the original query and we get the exception from the query with remarks.
+   * This function adjusts the value of the positions in the exception message to account for this.
+   * This is done in mildly scary kludge here in frontend after everything,
+   * because the alternative of doing it in backend
+   * is an absolutely terrifying kludge involving messing with exceptions.
+   */
+  let adjustmentLength = 0;
+
+  // redshift remarks use c-style multiline comments...
+  const multiLineBeginPos = origSql.search("/\\*");
+  const multiLineEndPos = origSql.search("\\*/");
+  // if multiLineBeginPos is 0 then we know it's a redshift remark
+  if (multiLineBeginPos === 0 && multiLineEndPos !== -1) {
+    adjustmentLength += multiLineEndPos + 2; // 2 for */ in itself
+  }
+
+  const chompedSql = origSql.substr(adjustmentLength);
+  // there also seem to be cases where remarks don't get in...
+  const commentPos = chompedSql.search("--");
+  const newLinePos = chompedSql.search("\n");
+  if (commentPos !== -1 && commentPos < 5) {
+    adjustmentLength += newLinePos + 2; // 2 for \n itself
+  }
```

Comment:
Why `< 5`?

Reply:
this is a heuristic so I should mark it as such. `remarks` gets added before and is a long comment ending with a newline. sometimes there is a space before the initial `--`, sometimes there is not. sometimes this one gets variegated with the redshift comments, which start with `/*` instead of `--`, but even with the occasional null remark which appears in redshift comments if it's before any putative `/**/` then you know it's a remark comment

Reply:
redshift is also why not to grep for the explicit `METABASE` yadda yadda